provider/core/src/key.rs

```diff
@@ -40,17 +41,26 @@ macro_rules! tagged {
 }
 
 /// A compact hash of a [`DataKey`]. Useful for keys in maps.
+///
+/// The hash will be stable over time within major releases.
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone, Hash, ULE)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 #[repr(transparent)]
 pub struct DataKeyHash([u8; 4]);
 
 impl DataKeyHash {
-    const fn compute_from_str(path: &str) -> Self {
-        Self(
-            helpers::fxhash_32(path.as_bytes(), leading_tag!().len(), trailing_tag!().len())
-                .to_le_bytes(),
-        )
+    const fn compute_from_path(path: &DataKeyPath) -> Self {
+        let hash = helpers::fxhash_32(
+            path.tagged.as_bytes(),
+            leading_tag!().len(),
+            trailing_tag!().len(),
+        );
+        Self(hash.to_le_bytes())
+    }
+
+    /// Gets the hash value as a byte array.
+    pub const fn to_bytes(&self) -> [u8; 4] {
```

Comment:
This is only used in a doc comment. We don't have to expose the hash internals, so we shouldn't.

Reply:
They are exposed already via ULE, and it needs to be a guarantee that these are stable across version numbers. I see this more as a mechanism to enforce that. Since we need this property, having this method is automatic.

Reply:
Note: the doc example for this function is below